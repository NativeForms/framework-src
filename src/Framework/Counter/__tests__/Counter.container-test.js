import React from 'react';
import Immutable from 'immutable';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';

// framework
import CounterContainer from '../Counter.container';
import initialState from '../Counter.state';

const mockStore = configureMockStore([]);

test('CounterContainer should render correctly', () => {
  const wrapper = shallow(<CounterContainer />, {
    context: { store: mockStore(Immutable.Map({ counter: initialState })) }
  });
  expect(wrapper.dive()).toMatchSnapshot();
  wrapper.setContext({
    context: { store: mockStore(Immutable.Map({ counter: Immutable.fromJS({ counter: 1 }) })) }
  });
  expect(wrapper.dive()).toMatchSnapshot();
});
