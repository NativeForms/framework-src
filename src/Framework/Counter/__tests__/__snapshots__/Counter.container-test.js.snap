// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterContainer should render correctly 1`] = `
<IntlProvider
  messages={
    Object {
      "clickedMessage": "Clicked: {count, plural, one {# time} other {# times}}",
      "incrementLabel": "Increment {type, select, odd {if odd} async {async}}",
      "title": "Counter",
    }
  }
>
  <Component
    counter={
      Immutable.Map {
        counter: 0,
      }
    }
    decrement={[Function]}
    increment={[Function]}
    incrementAsync={[Function]}
    incrementIfOdd={[Function]}
  />
</IntlProvider>
`;

exports[`CounterContainer should render correctly 2`] = `
<IntlProvider
  messages={
    Object {
      "clickedMessage": "Clicked: {count, plural, one {# time} other {# times}}",
      "incrementLabel": "Increment {type, select, odd {if odd} async {async}}",
      "title": "Counter",
    }
  }
>
  <Component
    counter={
      Immutable.Map {
        counter: 0,
      }
    }
    decrement={[Function]}
    increment={[Function]}
    incrementAsync={[Function]}
    incrementIfOdd={[Function]}
  />
</IntlProvider>
`;
